extendr-macros/src/extendr_impl.rs

```diff
@@ -187,86 +188,45 @@ pub fn extendr_impl(mut item_impl: ItemImpl, opts: &ExtendrOptions) -> syn::Resu
-                fn try_from(robj: &Robj) -> Result<Self> {
-                    use libR_sys::R_ExternalPtrAddr;
-                    unsafe {
-                        let ptr = R_ExternalPtrAddr(robj.get()).cast::<#self_ty>();
-                        ptr.as_ref().ok_or_else(|| Error::ExpectedExternalNonNullPtr(robj.clone()))
-                    }
-                }
-            }
+        impl TryFrom<Robj> for &mut #self_ty {
+            type Error = Error;
 
-            // Input conversion function for a mutable reference to this type.
-            impl TryFrom<&mut Robj> for &mut #self_ty {
-                type Error = Error;
-                fn try_from(robj: &mut Robj) -> Result<Self> {
-                    use libR_sys::R_ExternalPtrAddr;
-                    unsafe {
-                        let ptr = R_ExternalPtrAddr(robj.get_mut()).cast::<#self_ty>();
-                        ptr.as_mut().ok_or_else(|| Error::ExpectedExternalNonNullPtr(robj.clone()))
-                    }
-                }
+            fn try_from(mut robj: Robj) -> Result<Self> {
+                Self::try_from(&mut robj)
             }
         }
-    } else {
-        quote! {
-            // Input conversion function for this type.
-            impl<'a> extendr_api::FromRobj<'a> for &#self_ty {
-                fn from_robj(robj: &'a Robj) -> std::result::Result<Self, &'static str> {
-                    if robj.check_external_ptr_type::<#self_ty>() {
-                        #[allow(clippy::transmute_ptr_to_ref)]
-                        Ok(unsafe { std::mem::transmute(robj.external_ptr_addr::<#self_ty>()) })
-                    } else {
-                        Err(concat!("expected ", #self_ty_name))
-                    }
-                }
-            }
 
-            // Input conversion function for a reference to this type.
-            impl<'a> extendr_api::FromRobj<'a> for &mut #self_ty {
-                fn from_robj(robj: &'a Robj) -> std::result::Result<Self, &'static str> {
-                    if robj.check_external_ptr_type::<#self_ty>() {
-                        #[allow(clippy::transmute_ptr_to_ref)]
-                        Ok(unsafe { std::mem::transmute(robj.external_ptr_addr::<#self_ty>()) })
-                    } else {
-                        Err(concat!("expected ", #self_ty_name))
-                    }
+        // Output conversion function for this type.
+        impl TryFrom<&Robj> for &#self_ty {
+            type Error = Error;
+            fn try_from(robj: &Robj) -> Result<Self> {
+                use libR_sys::R_ExternalPtrAddr;
+                unsafe {
+                    let ptr = R_ExternalPtrAddr(robj.get()).cast::<#self_ty>();
+                    ptr.as_ref().ok_or_else(|| Error::ExpectedExternalNonNullPtr(robj.clone()))
                 }
             }
+        }
 
-            // Output conversion function for this type.
-                impl<'a> From<&'a #self_ty> for Robj {
```

Comment:
This should be documented in the changelog. This might be a breaking change? 

It is impossible to create an `Robj` from a reference now. Thist make sense conceptually. Allowing it previously, was an error on our part.

Reply:
I think I've covered in the changelog now.